cli/cmd/inject_test.go

```diff
@@ -177,6 +177,12 @@ func TestUninjectAndInject(t *testing.T) {
 			reportFileName:   "inject_emojivoto_deployment.report",
 			testInjectConfig: noInitContainerConfig,
 		},
+		{
+			inputFileName:    "inject_emojivoto_deployment_config_overrides.input.yml",
+			goldenFileName:   "inject_emojivoto_deployment_config_overrides.golden.yml",
+			reportFileName:   "inject_emojivoto_deployment.report",
+			testInjectConfig: defaultConfig,
+		},
```

Comment:
Oooh, we could also use these fixtures in uinject_test.go to validate that the `config.linkerd.io` annotations are preserved when uninjecting.

```diff
diff --git a/cli/cmd/uninject_test.go b/cli/cmd/uninject_test.go
index e7fdb7c7..91a7bb2c 100644
--- a/cli/cmd/uninject_test.go
+++ b/cli/cmd/uninject_test.go
@@ -63,6 +63,11 @@ func TestUninjectYAML(t *testing.T) {
 			goldenFileName: "inject_emojivoto_deployment.input.yml",
 			reportFileName: "inject_emojivoto_deployment_uninject.report",
 		},
+		{
+			inputFileName:  "inject_emojivoto_deployment_config_overrides.golden.yml",
+			goldenFileName: "inject_emojivoto_deployment_config_overrides.input.yml",
+			reportFileName: "inject_emojivoto_deployment_uninject.report",
+		},
 		{
 			inputFileName:  "inject_emojivoto_pod_tls.golden.yml",
 			goldenFileName: "inject_emojivoto_pod.input.yml",
```

But for that to work, you have to tweak the annotations in the input file slightly, as follows:

```diff
diff --git a/cli/cmd/testdata/inject_emojivoto_deployment_config_overrides.input.yml b/cli/cmd/testdata/inject_emojivoto_deployment_config_overrides.input.yml
index 5f67c697..4cd6bf59 100644
--- a/cli/cmd/testdata/inject_emojivoto_deployment_config_overrides.input.yml
+++ b/cli/cmd/testdata/inject_emojivoto_deployment_config_overrides.input.yml
@@ -12,17 +12,17 @@ spec:
   strategy: {}
   template:
     metadata:
+      annotations:
+        config.linkerd.io/metrics-port: "9998"
+        config.linkerd.io/proxy-cpu-limit: "1"
+        config.linkerd.io/proxy-cpu-request: "0.5"
+        config.linkerd.io/proxy-memory-limit: 256Mi
+        config.linkerd.io/proxy-memory-request: 64Mi
+        config.linkerd.io/skip-inbound-ports: 7777,8888
+        config.linkerd.io/skip-outbound-ports: "9999"
       creationTimestamp: null
       labels:
         app: web-svc
-      annotations:
-        config.linkerd.io/skip-inbound-ports: 7777,8888
-        config.linkerd.io/skip-outbound-ports: 9999
-        config.linkerd.io/metrics-port: 9998
-        config.linkerd.io/proxy-cpu-limit: 1.0
-        config.linkerd.io/proxy-cpu-request: 0.5
-        config.linkerd.io/proxy-memory-request: 64Mi
-        config.linkerd.io/proxy-memory-limit: 256Mi
     spec:
       containers:
       - env:
```